build/Dockerfile

```diff
@@ -12,58 +12,50 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM registry.access.redhat.com/ubi9/python-311 as wheel
 
-ARG WHEEL_VERSION=""
-USER root
-RUN --mount=type=cache,target=/root/.cache/pip \
-    python -m pip install --upgrade pip && \
-    python -m pip install build
-COPY tuning tuning
-COPY .git .git
-COPY pyproject.toml pyproject.toml
-# build wheel if wheel version is empty else download the wheel from PyPi
-RUN if [[ -z "${WHEEL_VERSION}" ]]; \
-    then python -m build --wheel --outdir /tmp; \
-    else pip download fms-hf-tuning==${WHEEL_VERSION} --dest /tmp --only-binary=:all: --no-deps; \
-    fi && \
-    ls /tmp/*.whl >/tmp/bdist_name
-
-
-FROM registry.access.redhat.com/ubi9/ubi AS release
-
-ARG CUDA_VERSION=11.8.0
+## Global Args #################################################################
+ARG BASE_UBI_IMAGE_TAG=latest
 ARG USER=tuning
 ARG USER_UID=1000
-ARG SET_NUM_PROCESSES_TO_NUM_GPUS=True
+ARG PYTHON_VERSION=3.11
+ARG WHEEL_VERSION=""
+
+## Base Layer ##################################################################
+FROM registry.access.redhat.com/ubi9/ubi:${BASE_UBI_IMAGE_TAG} as base
 
-USER root
+ARG PYTHON_VERSION
+ARG USER
+ARG USER_UID
 
 RUN dnf update -y \
-    && dnf remove -y --disableplugin=subscription-manager \
-        subscription-manager \
-        # we install newer version of requests via pip
-        python3.11-requests \
-    && dnf install -y make \
-        # to help with debugging
-        procps \
```

Comment:
procps is the `ps -ef` command for listing current running processes. I didn't see `ps` or `make` used by anything. I don't problem adding them back because they are useful for development.

Alternatively, we can add additional stage to install any dev tools
```
...
FROM release as devleopment
dnf install -y make procps git 
```
For release image, `docker build --targt=release -t sft-trainer:release  .  -f build/Dockerfile`
For developmen image `docker build --targt=development -t sft-trainer:dev .  -f build/Dockerfile`

Reply:
I agree that a development image would be helpful. It should also contain the fms-hf-tuning library and all deps needed

Reply:
I wonder if `ps` is okay to leave in the final image if it is helpful for seeing multi-GPU processes running and such